### PR TITLE
chore: Migrate from Create React App to Vite

### PR DESCRIPTION
**Pull request by janosbabik**
## Description

This change updates the frontend application to be generated with Vite instead of Create React App (CRA). 
The benefits include:
- Reduced build time
- Zero security vulnerabilities

image

Learn more about Vite: [https://vitejs.dev](https://vitejs.dev)

## Motivation and Context

The Create React App project is somewhat abandoned, lacking maintainers. Vite represents the next generation of frontend tooling, staying up-to-date. Additionally, the frontend build time is currently lengthy, and this migration to Vite aims to significantly reduce it.

## How Has This Been Tested

- Manually
- E2E

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/frontend/vite.config.mts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import react from '@vitejs/plugin-react';
+import { defineConfig, splitVendorChunkPlugin } from 'vite';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [splitVendorChunkPlugin(), react()],
+  server: {
+    open: true,
+    host: true,
+    port: 3000,
```
Comment (martin-trajanovski) [inline apps/frontend/vite.config.mts:10]: I was just thinking if the port needs to be hardcoded to 3000 or it is better to use an env variable.

File: apps/frontend/vite.config.mts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      ...[
+        '/graphql',
+        '/downloads',
+        '/download',
+        '/uploads',
+        '/files',
+        '/preview',
+      ].reduce((acc, curr) => {
+        acc[curr] = {
+          target: 'http://localhost:4000',
```
Comment (martin-trajanovski) [inline apps/frontend/vite.config.mts:21]: The same goes here. Maybe it is better to read this from environment variable first and have fallback to `http://localhost:4000` if that's not set.

**Review by martin-trajanovski — APPROVED** [review]
This looks like a great improvement. I just left some small comment that might be worth checking. Otherwise it's ✅

File: apps/frontend/package.json [hunk 3]
```diff
@@ -56,31 +56,26 @@
...
-    "test:debug": "react-scripts --inspect-brk test --runInBand --no-cache",
+    "dev": "vite",
+    "dev:docker": "npm install && vite",
+    "build": "vite build",
+    "preview": "vite preview",
     "lint": "tsc --noEmit && eslint . --ext .js,.jsx,.ts,.tsx --quiet",
     "lint:fix": "tsc --noEmit && eslint . --ext .js,.jsx,.ts,.tsx --fix",
     "lint:fix:bash": "eslint --fix $(git diff --name-only --diff-filter=ACMRTUXB develop | grep  -E \"(.js$|.ts$|.tsx$)\")",
-    "eject": "react-scripts eject",
     "start": "npm run dev",
```
Comment (enthusiastio) [inline apps/frontend/package.json:75]: AFAIK we are not using the `start` script, might be a good chance to clean it up,

**Review by enthusiastio — APPROVED** [review]

**Comment by janosbabik** [comment]
Thanks @martin-trajanovski , @enthusiastio for the comments. I have updated the code based on your feedback. Please let me know if there is anything else.

**Pull request merged**